ObjectCreation/AeroFoil.FCMacro

```diff
@@ -0,0 +1,1747 @@
+                App.ActiveDocument.getObjectsByLabel(
+                    "AeroFoil_" + str(refNum + i + 1).zfill(3)
+                )[0].Label = "AeroFoil_" + str(refNum + i + 1)
+                if self.designprogressbar.value() < 99:
+                    self.designprogressbar.setValue(
+                        round(75 + ((i / (4 * quantity_)) * 100))
+                    )
+        self.designprogressbar.setValue(100)
+        time.sleep(1)
+
+    def characterize(self):
+        try:
+            objRef = App.ActiveDocument.getObjectsByLabel(self.profileName)[0]
+            objRef.addProperty("App::PropertyString", "AirfoilType", "", "", 1)
+            objRef.addProperty("App::PropertyLength", "AirfoilChordLength", "", "", 1)
+            objRef.addProperty("App::PropertyString", "DesignCurveType", "", "", 1)
+            objRef.addProperty("App::PropertyInteger", "NumberOfPoints", "", "", 1)
+            tempStr = ""
+            if self.objRefData[0] == 1:
+                tempStr = "NACA - " + self.objRefData[1]
+            elif self.objRefData[0] == 2:
+                tempStr = "NACA - " + self.objRefData[2]
+            elif self.objRefData[0] == 3 and self.objRefData[3] == 1:
+                tempStr = "Custom Curves (Symmetric)"
+            elif self.objRefData[0] == 3 and self.objRefData[3] == 2:
+                tempStr = "Custom Curves (Asymmetric)"
+            elif self.objRefData[0] == 4 and self.objRefData[4] == 1:
+                tempStr = "Custom Points (DAT)"
+            elif self.objRefData[0] == 4 and self.objRefData[4] == 2:
+                tempStr = "Custom Points (CSV)"
+            objRef.AirfoilType = tempStr
+            objRef.AirfoilChordLength = str(self.chord) + "mm"
+            tempStr = ""
+            if self.designtype[1] == 1:
+                tempStr = "Polygon"
+            elif self.designtype[1] == 2:
+                tempStr = "BSpline"
+            objRef.DesignCurveType = tempStr
+            objRef.NumberOfPoints = self.n_units - 1
+        except Exception:
+            print(
+                "\nCannot generate AeroFoil properties in FreeCAD version &#60;0.19 !\n"
+            )
+            FreeCAD.Console.PrintWarning(
+                "\nCannot generate AeroFoil properties in FreeCAD version &#60;0.19 !\n"
+            )
+
+    def create(self):
+        constraintIndex = 0
+        # Generating Polygons and BSplines
+        try:
+            if self.designtype[0] == 2:
+                App.activeDocument().addObject(
+                    "Sketcher::SketchObject", self.profileName
+                )
+                App.ActiveDocument.getObject(
+                    self.profileName
+                ).Placement = App.Placement(
+                    App.Vector(0.000000, 0.000000, 0.000000),
+                    App.Rotation(-0.707107, 0.000000, 0.000000, -0.707107),
+                )
+                App.ActiveDocument.getObject(self.profileName).MapMode = "Deactivated"
```

Comment:
Getting an object by its name is not robust because you cannot call the macro twice. You'd rather "save" objects into Python variables and manipulate theses variables directly.

Reply:
Thank you, Mr. Galou. I've sorted that out now.